src/spend/public.rs

```diff
@@ -0,0 +1,151 @@
+use super::secret::SecretKey;
+use super::stealth::StealthAddress;
+
+use crate::sponge;
+use crate::{
+    decode::decode, Error, JubJubAffine, JubJubExtended, JubJubScalar,
+};
+
+use dusk_jubjub::GENERATOR;
+use std::convert::TryFrom;
+use std::fmt;
+use subtle::{Choice, ConstantTimeEq};
+
+/// Public pair of `a·G` and `b·G`
+#[derive(Debug, Clone, Copy)]
+pub struct PublicKey {
+    A: JubJubExtended,
+    B: JubJubExtended,
+}
+
+impl PublicKey {
+    /// This method is used to construct a new `PublicKey` from the given public
+    /// pair of `a·G` and `b·G`
+    pub fn new(A: JubJubExtended, B: JubJubExtended) -> Self {
+        Self { A, B }
+    }
+
+    /// Gets `A` (`a·G`)
+    pub fn A(&self) -> &JubJubExtended {
+        &self.A
+    }
+
+    /// Gets `B` (`b·G`)
+    pub fn B(&self) -> &JubJubExtended {
+        &self.B
+    }
+
+    /// Generates new `PKr = H(A · r) · G + B` from a given `r`
+    pub fn gen_stealth_address(&self, r: &JubJubScalar) -> StealthAddress {
+        let G = GENERATOR.to_niels();
```

Comment:
Any particular reason to use niels here?

Reply:
Because it's the idiomatic way to use the `GENERATOR` in the multiplication (in accordance with the jubjub crate) and it's consistent with the rest of our code base (@CPerezz used that as well in other repo).

Reply:
Just to perform the `GENERATOR` multiplication – since we have the same code in other part of our code base.
If that is not the way to go we probably should refactor everything in order to be consistent.